radio/src/gui/colorlcd/radio_theme.cpp

```diff
@@ -0,0 +1,723 @@
+  listBox->setLongPressHandler([=] (event_t event) {
+    auto menu = new Menu(listBox,false);
+
+    // you cant edit the default theme
+    menu->addLine(STR_ACTIVATE, [=]() {
+      tp->applyTheme(listBox->getSelected());
+      tp->setDefaultTheme(listBox->getSelected());
+      nameText->setTextFlags(COLOR_THEME_PRIMARY1);
+      authorText->setTextFlags(COLOR_THEME_PRIMARY1);
+      nameLabel->setTextFlags(COLOR_THEME_PRIMARY1);
+      authorLabel->setTextFlags(COLOR_THEME_PRIMARY1);
+      listBox->setActiveIndex(tp->getThemeIndex());
+    });
+
+    // you cant edit the default theme
+    if (listBox->getSelected() != 0) {
+      menu->addLine(STR_EDIT,
+        [=] () {
+          auto theme = tp->getThemeByIndex(currentTheme);
+          if (theme == nullptr) return;
+          
+          new ThemeEditPage(*theme, 
+            [=](ThemeFile &theme) {
+              auto curTheme = tp->getThemeByIndex(currentTheme);
+              if (curTheme != nullptr) {
+                curTheme->setName(theme.getName());
+                curTheme->setAuthor(theme.getAuthor());
+                curTheme->setInfo(theme.getInfo());
+
+                // update the colors that were edited
+                int n = 0;
+                for (auto color : theme.getColorList()) {
+                  curTheme->setColorByIndex(n, color.colorValue);
+                  n++;
+                }
+
+                // save it to disk so it will come back the next time.
+                curTheme->serialize();
+
+                // if the active theme was edited then activate it.
+                if (listBox->getSelected() == tp->getThemeIndex())
+                  tp->applyTheme(listBox->getSelected());
+
+                // the list of theme names might have changed
+                listBox->setNames(tp->getNames());
+              }
+            });
+      });
+    }
+
+    menu->addLine(STR_NEW, [=] () {
+      ThemeFile newTheme;
+
+      new ThemeDetailsDialog(window, newTheme, [=] (ThemeFile theme) {
+        if (strlen(theme.getName()) != 0) {
+          char path[NAME_LENGTH + 20];
+          strncpy(path, theme.getName(), NAME_LENGTH + 19);
+          removeAllWhiteSpace(path);
+          strncat(path, ".yml", NAME_LENGTH + 19);
+          theme.setPath(path);
+
+          // use the current themes color list to make the new theme
+          auto curTheme = tp->getCurrentTheme();
+          for (auto color : curTheme->getColorList())
+            theme.setColor(color.colorNumber, color.colorValue);
+
+          theme.serialize();
+          tp->refresh();
+          listBox->setNames(tp->getNames());
+          listBox->setSelected(currentTheme);
+        }
+      });
+    });
+
+    // you cant delete the default theme or the currently active theme
+    if (listBox->getSelected() != 0 && listBox->getSelected() != tp->getThemeIndex()) {
+      menu->addLine(STR_DELETE, [=] () {
+        if (confirmationDialog("Delete Theme: ", "Are you sure?")) {
```

Comment:
Can we add the theme name to the msg so we know which theme is being deleted - fixes PBKAC syndrome ;)